tests/expr/str/test-slice.py

```diff
@@ -0,0 +1,174 @@
+     slice(2, 7), slice(3, -1), slice(5, -8), slice(-8, 5), slice(None, -2)])
+def test_slice_normal(slice):
+    src = ["Hydrogen", "Helium", "Lithium", "Berillium", "Boron", "Carbon",
+           None, "Praseodymium"]
+    DT = dt.Frame(A=src)
+    RES = DT[:, f.A[slice]]
+    EXP = dt.Frame(A=[None if s is None else s[slice] for s in src])
+    assert_equals(RES, EXP)
+
+
+@pytest.mark.parametrize('slice',
+    [slice(None, None, 2), slice(None, None, 3), slice(2, None, 2),
+     slice(1, -1, 3), slice(-5, -1, 2), slice(None, -1, 2)])
+def test_slice_large_step(slice):
+    src = ["Mercury", "Venus", "Earth", "Mars", "Jupiter", "Saturn", "Uranus",
+           "Neptune"]
+    DT = dt.Frame(Planet=src)
+    RES = DT[:, f.Planet[slice]]
+    EXP = dt.Frame(Planet=[s[slice] for s in src])
+    assert_equals(RES, EXP)
+
+
+@pytest.mark.parametrize('slice',
+    [slice(None, None, -1), slice(None, None, -2), slice(2, None, -1),
+     slice(1, -1, -3), slice(-15, -2, -2), slice(None, -1, -2),
+     slice(None, 3, -2)])
+def test_slice_negative_step(slice):
+    src = ["Juneteenth", "Independence Day", "Christmas", "Halloween",
+           "Labor Day", "Memorial Day", "Native American Day", "MLK Day"]
+    DT = dt.Frame(Planet=src)
+    RES = DT[:, f.Planet[slice]]
+    EXP = dt.Frame(Planet=[s[slice] for s in src])
+    assert_equals(RES, EXP)
+
+
+@pytest.mark.parametrize('seed', [random.getrandbits(32) for _ in range(20)])
+def test_slice_unicode_random(seed):
+    random.seed(seed)
+    src = [
+        "По улиці вітер віє",
+        "Та сніг замітає.",
+        "По улиці попідтинню",
+        "Вдова шкандибає",
+        "Під дзвіницю, сердешная,",
+        "Руки простягати",
+        "До тих самих, до багатих,",
+        "Що сина в солдати",
+        "Позаторік заголили.",
+        "А думала жити...",
+        "Хоч на старість у невістки",
+        "В добрі одпочити.",
+        "Не довелось. Виблагала",
+        "Тую копійчину...",
+        "Та пречистій поставила",
+        "Свічечку за сина.",
+        "",
+        "møøse",
+        "𝔘𝔫𝔦𝔠𝔬𝔡𝔢",
+        "J̲o̲s̲é̲",
+        "🚑🐧💚💥✅",
+        "[Ⅰ] 💥, [Ⅱ] 🐕, [Ⅲ] 🦊, [Ⅳ] 🐽"
+    ]
+    a = None if random.random() < 0.2 else \
+        random.randint(-3, 10)
+    b = None if random.random() < 0.5 else \
+        random.randint(-8, 0) if random.random() < 0.6 else \
+        random.randint(15, 25)
+    c = None if random.random() < 0.6 else \
+        random.randint(2, 5) if random.random() < 0.5 else \
+        random.randint(-4, -1)
+    DT = dt.Frame(src)
+    RES = DT[:, f[0][a:b:c]]
+    EXP = dt.Frame([s[a:b:c] for s in src])
+    assert_equals(RES, EXP)
+
+
+def test_slice_expression():
+    DT = dt.Frame(A=["Monday", "Tuesday", "Wednesday", "Thursday", "Friday"],
+                  s=[0, 3, 2, 1, -4],
+                  e=[5, 4, 0, -1, None])
+    RES = DT[:, (f.A)[f.s:f.e]]
+    EXP = dt.Frame(A=["Monda", "s", "", "hursda", "iday"])
+    assert_equals(RES, EXP)
```

Comment:
May be add a test that the `void` column type is supported in expressions?

Reply:
done